File: backend/Models/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid.Models;

public class Question
{
    [Key]
    public int Id { get; set; }
    [ForeignKey(nameof(Quiz))]
    public int QuizId { get; set; }
    public int Order { get; set; }
    public string Body { get; set; } = null!;

    public ICollection<Answer> Answers { get; set; } = new HashSet<Answer>();
    public ICollection<Solution> Solutions { get; set; } = new HashSet<Solution>();
    public Quiz Quiz { get; set; } = null!;

    [NotMapped]
    public DateTimeOffset? LastAnswerTimestamp { get; set; }

    [NotMapped]
    public Answer? LastAnswer { get; set; }

    [NotMapped]
    public int[] QuestionIds { get; set; } = null!;

    public Question() { }
}

File: frontend/src/app/models/question.ts
import { format } from "date-fns"
import { Answer } from "./answer"
import { Solution } from "./solution"
import { Type } from "class-transformer"
import { Quiz } from "./quiz"

export class Question {
    id?: number
    quizId?: number
    order?: number
    body?: string

    @Type(() => Date)
    lastAnswerTimestamp?: Date
    lastAnswer?: Answer
    questionIds?: number[]

    solutions: Solution[] = []
    answers: Answer[] = []
    quiz?: Quiz

    get formatLastAnswerTimestamp(): string {
        return `${this.lastAnswerTimestamp ? format(this.lastAnswerTimestamp!, 'dd/MM/yyyy HH:mm:ss') : ''}`;
    }

    get lastAnswerSql(): string {
        if (this.answers?.length > 0) {
            const lastAnswer = this.lastAnswer
            return `${lastAnswer?.sql ? lastAnswer.sql : ''}`;
        } else {
            return ''
        }
    }

    get databaseName(): string {
        return this.quiz!.database!.name!
    }
}